[beta] 1.79 promotion

Mark-Simulacrum opened the pull request:
https://forge.rust-lang.org/release/process.html#promote-branches-t-3-days-monday

r? @Mark-Simulacrum

Mark-Simulacrum committed:
Replace version placeholders for 1.79

Mark-Simulacrum committed:
Bump to beta

rust-log-analyzer commented:
The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8867972457/job/24347060161) [(plain)](https://github.com/rust-lang/rust/commit/a553f4262edb1e5431a50dd8ede00350f0751311/checks/24347060161/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
  CACHE_DOMAIN: ci-caches.rust-lang.org
  PR_CI_JOB: 1
  IMAGE: mingw-check
##[endgroup]
fatal: unknown commit 53fd98ca776cb875bc9e5514f56b52eb74f9e7a9
All commits in `HEAD` are present in `master`
src/ci/scripts/verify-stable-version-number.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
 Documenting rustc_data_structures v0.0.0 (/checkout/compiler/rustc_data_structures)
    Checking rustc_infer v0.0.0 (/checkout/compiler/rustc_infer)
    Checking rustc_metadata v0.0.0 (/checkout/compiler/rustc_metadata)
    Checking rustc_mir_dataflow v0.0.0 (/checkout/compiler/rustc_mir_dataflow)
error[E0080]: evaluation of `<std::cell::RefMut<'_, [u8]> as zerocopy::ByteSlice>::INTO_REF_INTO_MUT_ARE_SOUND` failed
    --> /cargo/registry/src/index.crates.io-6f17d22bba15001f/zerocopy-0.7.32/src/lib.rs:5238:9
     |
5238 | ...   panic!("Ref::into_ref and Ref::into_mut are unsound when used with core::cell::RefMut; see https://github.com/google/zerocopy/issues/716")
     |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'Ref::into_ref and Ref::into_mut are unsound when used with core::cell::RefMut; see https://github.com/google/zerocopy/issues/716', /cargo/registry/src/index.crates.io-6f17d22bba15001f/zerocopy-0.7.32/src/lib.rs:5238:9
     = note: this error originates in the macro `$crate::panic::panic_2015` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)


error[E0080]: evaluation of `<std::cell::Ref<'_, [u8]> as zerocopy::ByteSlice>::INTO_REF_INTO_MUT_ARE_SOUND` failed
    --> /cargo/registry/src/index.crates.io-6f17d22bba15001f/zerocopy-0.7.32/src/lib.rs:5218:9
     |
5218 | ...   panic!("Ref::into_ref and Ref::into_mut are unsound when used with core::cell::Ref; see https://github.com/google/zerocopy/issues/716")
     |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'Ref::into_ref and Ref::into_mut are unsound when used with core::cell::Ref; see https://github.com/google/zerocopy/issues/716', /cargo/registry/src/index.crates.io-6f17d22bba15001f/zerocopy-0.7.32/src/lib.rs:5218:9
     = note: this error originates in the macro `$crate::panic::panic_2015` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0080`.
error: could not document `rustc_data_structures`
```

</details>

Mark-Simulacrum committed:
Mark more entries in rustc_data_structures as no_inline for docs

This is a workaround for #122758, but it's not clear why 1.79 requires a
more extensive amount of no_inline than the previous release. Seems like
there's something relatively subtle happening here.

Mark-Simulacrum commented:
@bors r+ rollup=never p=50

bors commented:
:pushpin: Commit 532195dd7198e33a9adbbb487cdfa3d450bf0093 has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum 532195dd7198e33a9adbbb487cdfa3d450bf0093 -->
<!-- homu: {"type":"Approved","sha":"532195dd7198e33a9adbbb487cdfa3d450bf0093","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 532195dd7198e33a9adbbb487cdfa3d450bf0093 with merge 6b544f5ff8d45221d61962651a5f5ab9fe535e16...
<!-- homu: {"type":"BuildStarted","head_sha":"532195dd7198e33a9adbbb487cdfa3d450bf0093","merge_sha":"6b544f5ff8d45221d61962651a5f5ab9fe535e16"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/8869142981/job/24350922003)
Approved by: Mark-Simulacrum
Pushing 6b544f5ff8d45221d61962651a5f5ab9fe535e16 to beta...
<!-- homu: {"type":"BuildCompleted","approved_by":"Mark-Simulacrum","base_ref":"beta","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/8869142981/job/24350922003"},"merge_sha":"6b544f5ff8d45221d61962651a5f5ab9fe535e16"} -->
